yt/visualization/geo_plot_utils.py

```diff
@@ -0,0 +1,62 @@
+valid_transforms={}
+
+transform_list = ['PlateCarree', 'LambertConformal', 'LambertCylindrical',
+                  'Mercator', 'Miller', 'Mollweide', 'Orthographic',
+                  'Robinson', 'Stereographic', 'TransverseMercator',
+                  'InterruptedGoodeHomolosine', 'RotatedPole', 'OSGB',
+                  'EuroPP', 'Geostationary', 'Gnomonic', 'NorthPolarStereo',
+                  'OSNI', 'SouthPolarStereo', 'AlbersEqualArea',
+                  'AzimuthalEquidistant', 'Sinusoidal', 'UTM',
+                  'NearsidePerspective', 'LambertAzimuthalEqualArea']
+
+def cartopy_importer(transform_name):
+    r"""Convenience function to import cartopy projection types
+    """
+
+    def _func(*args, **kwargs):
+        from yt.utilities.on_demand_imports import _cartopy as cartopy
+        return getattr(cartopy.crs, transform_name)(*args, **kwargs)
+    return _func
+
+def get_mpl_transform(mpl_proj):
```

Comment:
Ok, i've modified this functionality to accept either a string, tuple or an instance of a cartopy.crs.CRS object. I think this should make it easy for users to access the transforms and projections in a few different ways. 

I'm not sure I understand what you're getting at by:
> If you really want string based lookups, then avoiding duplication would be a nice option.... getattr('cartopy.crs', transform_name).

Am I duplicating things with the way I wrote it in l18? This might be a subtle thing I'm not understanding.

Reply:
Thank you so much for your review @pelson ! Your feeback really improved my PR. 

I'd like to test out passing user-defined projections (other than the ones already available in cartopy) into the stuff I wrote, but I couldn't find any examples in the cartopy documentation. Could you perhaps point me to a working example of somebody using one somewhere that I can experiment with?